From 34a94688b830ee91f62dbf6af03fba0beb46db0d Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Wed, 18 Jun 2014 20:04:21 -0400
Subject: [block] blk_max_size_offset() should check ->max_sectors

Message-id: <1403121864-7956-5-git-send-email-snitzer@redhat.com>
Patchwork-id: 84269
O-Subject: [RHEL7.1 PATCH 155/150] block: blk_max_size_offset() should check ->max_sectors
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 736ed4de766d4f0e8e6142dd4f9d73ef61835ed9
Author: Jens Axboe <axboe@fb.com>
Date:   Tue Jun 17 22:09:29 2014 -0700

    block: blk_max_size_offset() should check ->max_sectors

    Commit 762380ad9322 inadvertently changed a check for max_sectors
    to max_hw_sectors. Revert that part, so we still compare against
    max_sectors.

    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 49f4bf0..20e649b 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -966,7 +966,7 @@ static inline unsigned int blk_max_size_offset(struct request_queue *q,
             sector_t offset)
 {
  if (!q->limits.chunk_sectors)
-  return q->limits.max_hw_sectors;
+  return q->limits.max_sectors;
 
  return q->limits.chunk_sectors -
    (offset & (q->limits.chunk_sectors - 1));
-- 
1.7.1